azkaban-web-server/src/main/java/azkaban/imagemgmt/servlets/ImageRampupServlet.java

```diff
@@ -0,0 +1,226 @@
+ * are the supported APIs. Create Image Rampup API: POST /imageRampup?session.id=? --data
+ *
+ * @payload.json Get active rampup for an image type : GET /imageRampup/spark_job?session.id=?
+ * Update Image Rampup API: POST /imageRampup/spark_job?session.id=? --data @payload.json
+ */
+public class ImageRampupServlet extends LoginAbstractAzkabanServlet {
+
+  private static final String BASE_IMAGE_RAMPUP_URI = "/imageRampup";
+  private static final String IMAGE_RAMPUP_PLAN_ID_KEY = "id";
+  private static final UriTemplate CREATE_IMAGE_RAMPUP_URI_TEMPLATE = new UriTemplate(
+      String.format("/imageRampup/{%s}", IMAGE_RAMPUP_PLAN_ID_KEY));
+  private static final UriTemplate UPDATE_IMAGE_RAMPUP_URI_TEMPLATE = new UriTemplate(
+      String.format("/imageRampup/{%s}", IMAGE_TYPE));
+  private ImageRampupService imageRampupService;
+  private ObjectMapper objectMapper;
+
+  private static final Logger log = LoggerFactory.getLogger(ImageRampupServlet.class);
+
+  public ImageRampupServlet() {
+    super(new ArrayList<>());
+  }
+
+  @Override
+  public void init(final ServletConfig config) throws ServletException {
+    super.init(config);
+    final AzkabanWebServer server = (AzkabanWebServer) getApplication();
+    this.objectMapper = server.getObjectMapper();
+    this.imageRampupService = server.getImageRampupService();
+  }
+
+  @Override
+  protected void handleGet(
+      final HttpServletRequest req, final HttpServletResponse resp, final Session session)
+      throws ServletException, IOException {
+    /* Get specific record */
+    try {
+      String response = null;
+      if (BASE_IMAGE_RAMPUP_URI.equals(req.getRequestURI())) {
+        // imageType must present. If not present throws ImageMgmtValidationException
+        final String imageType = HttpRequestUtils.getParam(req, IMAGE_TYPE);
+        if (imageType == null) {
+          log.error("Image type can't be null. Must provide valid image type to get active rampup"
+              + " plan");
+          throw new ImageMgmtValidationException(
+              "Image type can't be null. Must provide valid image type to get rampup plan.");
+        }
+        // Check for required permission to invoke the API
+        if (!hasPermission(imageType, session.getUser(), Type.GET)) {
+          throw new ImageMgmtInvalidPermissionException("Invalid permission to get image rampup "
+              + "plan");
+        }
+
+        // invoke service method and get response in string format
+        final Optional<ImageRampupPlan> imageRampupPlan =
+            this.imageRampupService.getActiveRampupPlan(imageType);
+        if (imageRampupPlan.isPresent()) {
+          response = this.objectMapper.writerWithDefaultPrettyPrinter()
+              .writeValueAsString(imageRampupPlan.get());
+        } else {
+          throw new ImageMgmtException(
+              String.format("There is no active rampup plan found for image "
+                  + "type: %s.", imageType));
+        }
```

Comment:
Filtering (`GET /imageRampup`) should return a list of matches if there is no result then we should return an empty list but not an error.
If on the other hand the user uses the specific resource URI (/imageRampup/{id}) then we can return a 404. 

The fact that there can only be one active plan per image it's irrelevant here, when searching/filtering we should always return a list. Could you please add the results from the service to a list before returning to the user.